### PR TITLE
Remove misleading statement about macOS support

### PR DESCRIPTION
**Pull request by kmilos**

### CODE REVIEW

**Review by zisoft — APPROVED** [review]
It was in fact too early to drop support for macOS versions with the darktable summer release, because every new macOS version is always released in October. So the darktable winter release would be the time to drop.

We should keep that in mind for 2025 and macOS 16.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -233,11 +233,9 @@ changes (where available).
   `darktable.css` file on your system. See
   https://github.com/darktable-org/darktable/issues/13166.
 
-- Starting with release 4.4 a new support policy regarding macOS
-  versions has been put in place – darktable releases will now only
-  support those macOS versions that are also supported by Apple.
-  Release 4.8 therefore drops support for macOS versions older than
-  13.5.
+- Starting with release 4.4 a new support policy regarding macOS versions
```
Comment (victoryforce) [inline RELEASE_NOTES.md:236]: So what is this policy? If we write that we have a policy, then we need to disclose it.

Comment (zisoft) [reply RELEASE_NOTES.md:236]: The policy was to support only macOS versions which are still supported by Apple.

Comment (victoryforce) [reply RELEASE_NOTES.md:236]: >> So what is this policy? If we write that we have a policy, then we need to disclose it.
> The policy was to support only macOS versions which are still supported by Apple.

Yes, but this is exactly the explanation that this PR removes. This PR mentions some policy but does not disclose it (in addition, the policy is not actually followed). This only creates confusion for the user.

> For 4.8 we didn't really follow it, so this is unfortunately just a statement of facts now.

The statement of facts is that we no longer support macOS versions older than 13.5. But why mention the policy, if in reality the policy is not followed?

`Release 4.8 drops support for macOS versions older than 13.5.` and nothing more should be added.

**Review by MStraeten — APPROVED** [review]

**Pull request merged**